### PR TITLE
Clean `config` parameters

### PR DESCRIPTION
**Pull request by wthrajat**
### Aims to fix https://github.com/citadel-tech/coinswap/issues/248

- [x] Removed internal parameters from `MakerConfig` and `TakerConfig` that should not be exposed to users.
- [x] Hardcoded default values for internal parameters in `maker/server.rs`.
- [x] Hardcoded default values for internal parameters in `taker/api.rs`.
- [x] Replaced separate `directory_server_onion_address` and `directory_server_clearnet_address` fields with a unified `directory_server_address` field for both `MakerConfig` and `TakerConfig`.
- [x] Set `required_confirms` to a default value of `1` for `MakerConfig`, removing it from user-facing configuration options.
- [X] Adjusted the code to use the constants rather than the struct fields.
- [x] Added documentation wherever needed.

### CODE REVIEW

File: src/maker/config.rs [hunk 1]
```diff
@@ -14,32 +14,20 @@ pub struct MakerConfig {
...
     pub absolute_fee_sats: Amount,
     /// Fee rate per swap amount in ppb.
     pub amount_relative_fee_ppb: Amount,
     /// Fee rate for timelocked contract in ppb
     pub time_relative_fee_ppb: Amount,
-    /// No of confirmation required for funding transaction
-    pub required_confirms: u64,
-    // Minimum timelock difference between contract transaction of two hops
+    /// Minimum timelock difference between contract transaction of two hops
     pub min_contract_reaction_time: u16,
```
Comment (mojoX911) [inline src/maker/config.rs:24]: Move this one to internal constants. We don't have enough security measures in the protocol to expose this safely to the user.

File: src/maker/config.rs [hunk 2]
```diff
@@ -53,20 +41,14 @@ impl Default for MakerConfig {
...
             time_relative_fee_ppb: Amount::from_sat(100_000),
-            required_confirms: 1,
             min_contract_reaction_time: 48,
             min_size: 10_000,
             socks_port: 19050,
-            directory_server_onion_address: "directoryhiddenserviceaddress.onion:8080".to_string(),
-            directory_server_clearnet_address: "127.0.0.1:8080".to_string(),
-            fidelity_value: 5_000_000, // 5 million  sats
+            directory_server_address: "directoryhiddenserviceaddress.onion:8080".to_string(),
+            fidelity_value: 5_000_000, // 5 million sats
```
Comment (mojoX911) [inline src/maker/config.rs:51]: General question. Is 5 mil sats a good default? 
cc @Shourya742

File: src/maker/config.rs [hunk 3]
```diff
@@ -14,32 +14,20 @@ pub struct MakerConfig {
...
-    /// Time interval ping directory server
-    pub directory_servers_refresh_interval_secs: u64,
-    /// Time interval to close a connection if no response is received
-    pub idle_connection_timeout: u64,
     /// Absolute coinswap fee
     pub absolute_fee_sats: Amount,
     /// Fee rate per swap amount in ppb.
     pub amount_relative_fee_ppb: Amount,
     /// Fee rate for timelocked contract in ppb
     pub time_relative_fee_ppb: Amount,
```
Comment (mojoX911) [inline src/maker/config.rs:22]: We need to explain what is PPB and how to calculate this value. Just a two lines of code comment is enough. 

I would suggest moving this to const for now and exposing it later in later releases. It's not a super important feature. In fact keeping it constant at 0, for initial public beta testing will make the swap fee calculations easier as well as user-facing configs simpler.

Later on we can always add it back as a "new cool feature". Better not to confuse the first users with it.

**Review by mojoX911 — CHANGES_REQUESTED** [review]
Awesome. Thanks for doing this cleanup.

Our configs looks much better simpler and saner now. 

Few more comments and its good to go.

File: src/maker/config.rs (hunk 2 above)
Comment (Shourya742) [reply src/maker/config.rs:51]: Can be lowered but works with our test..so no issue as of now..

File: src/maker/config.rs (hunk 1 above)
Comment (wthrajat) [reply src/maker/config.rs]: Done in https://github.com/citadel-tech/coinswap/commit/4237e14a6f8457e293117a63dd103a0e5c1c0d3f

**Review by Shourya742 — APPROVED** [review]

**Review by mojoX911 — APPROVED** [review]
Ack

**Pull request merged**